rewrite-properties/src/main/java/org/openrewrite/properties/trait/PropertiesReference.java

```diff
@@ -64,28 +62,8 @@ public Tree rename(Renamer renamer, Cursor cursor, ExecutionContext ctx) {
         return tree;
     }
 
-    private static class Matcher extends SimpleTraitMatcher<PropertiesReference> {
```

Comment:
What's the reason to inline the matcher here only? And do you expect any impact from creating a new anonymous class with Predicate versus having one static class with static field?

Reply:
The reason I did it has nothing to do with performance, but more with the intention of the code. The implementation of the matcher should have meaning outside of the Provider class. 

But you are right of course, doing it this way could potentially impact performance! The right way would have been to move static class inside the Provider class. I will improve it, thx!